Lib/test/libregrtest/main.py

```diff
@@ -331,6 +333,19 @@ def run_tests_sequentially(self, runtests) -> None:
 
             result = self.run_test(test_name, runtests, tracer)
 
+            # Unload the newly imported modules (best effort finalization)
+            new_modules = [module for module in sys.modules
+                           if module not in save_modules and
+                                module.startswith("test.")]
+            for module in new_modules:
+                sys.modules.pop(module, None)
+                # Remove the attribute of the parent module.
+                parent, _, name = module.rpartition('.')
```

Comment:
Maybe raise an error if the parent itself has a parent. Currently, it should not happen. But it may happen in the future, if tests are reorganized.